### PR TITLE
docs(aio): Added developer guide on RxJS

### PR DESCRIPTION
**Pull request by brandonroberts**
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Other... Please describe:
```

Developer guide on observable usage. This was migrated over from https://github.com/angular/angular.io/pull/3047

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: aio/content/navigation.json [hunk 1]
```diff
@@ -245,6 +245,18 @@
         },
 
         {
+          "url": "guide/rxjs",
+          "title": "Observables",
+          "tooltip": "Using Observables to manage event streams."
+        },
+
+        {
+          "url": "guide/pipes",
```
Comment (gkalpak) [inline aio/content/navigation.json]: Is it intentional that pipes are added here, while they are already present above (around line 188)?

Comment (brandonroberts) [inline aio/content/navigation.json]: I doubt it. Probably happened during a rebase.

**Review by Foxandxss — CHANGES_REQUESTED** [review]
Where is the plunker? Rxjs is a complex subject and there is a lot of examples that you don't know what kind of output to expect.

Continuing on that, there are a few simple snippets that could use some comments saying what is the expected output.

I saw you named hot and cold but there is no more on it. A link would be nice.

There is a part where you give a few resources link. That is better to be at the end. To not distract people from reading your guide.

I find the first part (Observable is a function) to be too complicated.

**Comment by jenniferfell** [comment]
@jasonaden :  Recommendation here? Has all of this been incorporated into the new doc you're working on, so this can be closed? Other next steps on this PR? Thanks.

**Comment by jenniferfell** [comment]
@jasonaden Can we close this as a duplicate of https://github.com/angular/angular/pull/21423 ?

**Comment by splincode** [comment]
@brandonroberts what is the status of your pull request? is it still relevant? please fix conflicts if they exist

**Comment by brandonroberts** [comment]
@splincode this guide has been folded into the Observables guide except for the marble testing area, which will come along later. I think this can be closed.

**Pull request closed without merging**